drivers/serial/uart_stm32.c

```diff
@@ -533,7 +533,8 @@ static int uart_stm32_irq_tx_ready(const struct device *dev)
 {
 	USART_TypeDef *UartInstance = UART_STRUCT(dev);
 
-	return LL_USART_IsActiveFlag_TXE(UartInstance);
+	return LL_USART_IsActiveFlag_TXE(UartInstance) &&
+		LL_USART_IsEnabledIT_TC(UartInstance);
```

Comment:
why mixing LL_USART_IsActiveFlag_TXE with TC here ?
as the `_uart_stm32_irq_tx_complete_` exists and could be:
LL_USART_IsActiveFlag_TC && LL_USART_IsEnabledIT_TC

Reply:
@FRASTM Agree this is not clear and situation is due to a reported behavior where `rq_tx_ready` could be called after `irq_tx_disable` has been called and in that case is "expected" to return 0.
This is not explicit in the API, but apparently this behavior is also used in other (at least some) uart drivers.
To achieve that, since `irq_tx_disable` is actually disabling TC, this is TC flag that should checked here.